frame/treasury/src/lib.rs

```diff
@@ -111,6 +111,16 @@ type BalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trai
 type PositiveImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::PositiveImbalance;
 type NegativeImbalanceOf<T> = <<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::NegativeImbalance;
 
+#[macro_export]
+macro_rules! auto_construct_runtime {
```

Comment:
Nice idea, as you already said, it would be nice to replace this with
```
construct_runtime_args!( Module, Call, Storage, Config,  Event<T> )
```

`construct_runtime_args` can be replaced with a better name.